src/roiextractors/extractors/memmapextractors/memmapextractors.py

```diff
@@ -64,43 +62,80 @@ def get_num_channels(self):
         return self._num_channels
 
     @staticmethod
-    def write_imaging(imaging_extractor: ImagingExtractor, save_path: PathType = None, verbose: bool = False):
+    def write_imaging(
+        imaging_extractor: ImagingExtractor,
+        save_path: PathType,
+        verbose: bool = False,
+        buffer_gb: int = 0,
```

Comment:
@h-mayorquin Minor detail - annotation type should be a `float`. It's also a little odd to default it to `0` (which I assume means no buffering). I'd think if `buffer_gb` is set to any value, then that should specify using iterative write. To disable iterative write, either set an artificial `buffer_gb > actual_data_size` or don't make use of iterative write code at all. The former would probably take less code, but in either case I think it makes more sense to set the default value here to `None` to indicate no iterative write.

```
        buffer_gb: Optional[float] = None,
```

Reply:
No, you are wrong. 0 means that you will write frame by frame. I think that this makes as much sense as using `None` to denote this. 

The iterative write is disabled with when the buffer is large enough as your specifying.

Reply:
> No, you are wrong. 0 means that you will write frame by frame.

Ah, I did not realize that writing frame-by-frame implies that the size of the buffer is not only less than the size of a frame, but in fact zero!

`None` communicates this 'special' behavior more clearly, since it is equivalent to performing an automatic calulation of `buffer_gb=dtype.itemsize*n_rows*n_cols*n_channels/1e9`, which is clearly not zero.

Reply:
All right, I am on the fence on this but you have more experience dealing with this object. The desired changes are in the last commit.